src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs

```diff
@@ -101,6 +101,18 @@ public class SqlClientInstrumentationOptions
         /// </remarks>
         public bool EnableConnectionLevelAttributes { get; set; }
 
+        /// <summary>
+        /// Gets or sets a Filter function that determines whether or not to collect telemetry about a command
+        /// The Filter gets the SqlCommand, and should return a boolean.
+        /// If Filter returns true, the request is collected.
+        /// If Filter returns false or throw exception, the request is filtered out.
+        /// </summary>
+        /// <remarks>
+        /// <para>object: the raw <c>SqlCommand</c> object to interrogate for a decision on whether to trace or not.</para>
+        /// </remarks>
+        /// <returns>true to collect request, false to filter out.</returns>
+        public Func<object, bool> Filter { get; set; }
```

Comment:
#if !NETFRAMEWORK as we are only adding support for .NET Core.

Reply:
@vishweshbankwar moved Filter section under the `#if !NETFRAMEWORK` clause